### PR TITLE
Changes to the Running MacrosGump

### PR DESCRIPTION
**Pull request by qkdefus**
Changes to the Running MacrosGump, replaced the background using alpharegion
Added an option to save and set heigh width for the gump since it was unable to show long macro names
Toggling "show gump" now closes the gump when not checked

TODO : update gump on height/width new value
TODO : add text color option ?

### CODE REVIEW

**Comment by qkdefus** [comment]
img3
img1
img2

**Comment by qkdefus** [comment]
looking at the picture now, i notice height width numbers doesnt match up :P

File: ClassicAssist/Data/Options.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public class Options : SetPropertyNotifyChanged
     {
         public const string DEFAULT_SETTINGS_FILENAME = "settings.json";
         private static string _profilePath;
-        private bool _abilitiesGump = true;
+        private bool _abilitiesGump = false;
```
Comment (Reetus) [inline ClassicAssist/Data/Options.cs:23]: Undo this

File: ClassicAssist/Data/Options.cs [hunk 2]
```diff
@@ -57,6 +57,8 @@ public class Options : SetPropertyNotifyChanged
         private bool _macrosGump;
         private int _macrosGumpX;
         private int _macrosGumpY;
+        private int _macrosGumpHeight = 500;
```
Comment (Reetus) [inline ClassicAssist/Data/Options.cs:60]: Make the default height/width the same as it is currently

File: ClassicAssist/UI/Views/OptionsTabControl.xaml.cs [hunk 3]
```diff
@@ -11,5 +11,18 @@ public OptionsTabControl()
         {
             InitializeComponent();
         }
+
+        private void TextBox_MacrosGumpHeight_TextChanged( object sender, TextChangedEventArgs e )
```
Comment (Reetus) [inline ClassicAssist/UI/Views/OptionsTabControl.xaml.cs:15]: Remove these event handlers if they're not used

File: ClassicAssist/UI/Views/OptionsTabControl.xaml.cs [hunk 4]
```diff
@@ -11,5 +11,18 @@ public OptionsTabControl()
...
+        {
+            //UO.Gumps.MacrosGump.Initialize();
+            //UO.Gumps.MacrosGump.ResendGump( true );
+            //UO.Gumps.MacrosGump.SetPosition(100, 100);
+        }
+
+        private void TextBox_MacrosGumpWidth_TextChanged( object sender, TextChangedEventArgs e )
```
Comment (Reetus) [inline ClassicAssist/UI/Views/OptionsTabControl.xaml.cs:22]: Remove these event handlers if they're not used

File: ClassicAssist/UO/Gumps/MacrosGump.cs [hunk 5]
```diff
@@ -35,19 +35,27 @@ public class MacrosGump : RepositionableGump
         private static IEnumerable<MacroEntry> _lastMacros;
         private readonly MacroEntry[] _macros;
 
-        public MacrosGump( IEnumerable<MacroEntry> macros ) : base( 190, 180, _serial++, (uint) _serial++ )
+        // TODO : read value from options tab
+        //private static int _sizeX = Options.CurrentOptions.MacrosGumpHeight;
```
Comment (Reetus) [inline ClassicAssist/UO/Gumps/MacrosGump.cs:39]: This comments appear no longer relevant, if so remove them

File: ClassicAssist/UO/Gumps/MacrosGump.cs [hunk 6]
```diff
@@ -35,19 +35,27 @@ public class MacrosGump : RepositionableGump
...
 
             GumpX = Options.CurrentOptions.MacrosGumpX;
             GumpY = Options.CurrentOptions.MacrosGumpY;
 
-            Movable = false;
-            Closable = false;
+            int _height = Options.CurrentOptions.MacrosGumpHeight;
+            int _width = Options.CurrentOptions.MacrosGumpWidth;
+
+            Movable = true;
```
Comment (Reetus) [inline ClassicAssist/UO/Gumps/MacrosGump.cs:52]: Set Movable and Closable back to false, if they move the gump around by dragging it, we won't know the position to save it

File: ClassicAssist/UO/Gumps/MacrosGump.cs [hunk 7]
```diff
@@ -59,15 +67,18 @@ public MacrosGump( IEnumerable<MacroEntry> macros ) : base( 190, 180, _serial++,
...
                 {
-                    html = $"<BASEFONT COLOR=#000000><I>{macro.Name}</I></BASEFONT>\n";
+                    html = $"<BASEFONT face=Arial color=red><I>{macro.Name}</I></BASEFONT>\n";
                 }
 
-                AddHtml( 20, y, 150, 140, html, false, false );
-                AddButton( 160, y + 3, 3, 4, i++, GumpButtonType.Reply, 0 );
+                AddHtml( 20, y, _width - 40, _height - 40, html, false, false );
+   
+                //AddButton( _sizeX - 30, y + 3, 2104, 2103, i++, GumpButtonType.Reply, 0, ElementType.button, 2, 2 );
```
Comment (Reetus) [inline ClassicAssist/UO/Gumps/MacrosGump.cs:79]: Remove if it doesn't do anything

File: ClassicAssist/UO/Objects/Gumps/Gump.cs [hunk 8]
```diff
@@ -125,6 +125,10 @@ public GumpElement[] GumpElements
 
         public bool Movable { get; set; }
 
+        //public int _sizeX { get; set; }
```
Comment (Reetus) [inline ClassicAssist/UO/Objects/Gumps/Gump.cs:128]: Remove if it doesn't do anything

File: ClassicAssist/UO/Objects/Gumps/Gump.cs [hunk 9]
```diff
@@ -483,6 +487,7 @@ public void AddButton( int x, int y, int normalID, int pressedID, int buttonID,
                 ButtonType = (int) type,
                 ElementID = buttonID,
                 Param = param
+
```
Comment (Reetus) [inline ClassicAssist/UO/Objects/Gumps/Gump.cs:490]: Remove the superfluous new line

**Comment by Reetus** [comment]
I will change MacrosGump and WeaponAbilitiesGump to try and use reflection to get the x,y of the gump on gump close / disconnected...

The PR is https://github.com/Reetus/ClassicAssist/pull/318

You can either reimplement your changes onto that branch, or I will do so once you've finished your edits

**Comment by qkdefus** [comment]
please figure out a better place to put the `_dispatcher.Invoke( () => { _selectedTab = SelectedTab; } );` 
i honestly cant figure it out how to do this properly :)

**Comment by Reetus** [comment]
ClassicAssist-2023-1-16-6-27-4

I've moved the selected tab changes to a different branch, it's jumping back to general for me and different changes should be in different PR's anyway

https://github.com/Reetus/ClassicAssist/tree/selected-tab

**Comment by qkdefus** [comment]
weird, the tab issue doesnt happen on my end.

**Pull request merged**